qt/scientific_interfaces/EnggDiffraction/EnggDiffGSASFittingModel.cpp

```diff
@@ -145,23 +148,34 @@ void EnggDiffGSASFittingModel::doRefinements(
   qRegisterMetaType<
       MantidQt::CustomInterfaces::GSASIIRefineFitPeaksOutputProperties>(
       "GSASIIRefineFitPeaksOutputProperties");
+  qRegisterMetaType<Mantid::API::IAlgorithm_sptr>("IAlgorithm_sptr");
+  qRegisterMetaType<std::vector<GSASIIRefineFitPeaksOutputProperties>>(
+      "std::vector<GSASIIRefineFitPeaksOutputProperties>");
```

Comment:
Instead of having these dotted all over the code base can we think of a sensible common place to put them?
For the algorithm registration the common place seems like [line `227` in `MantidUI.cpp`](https://github.com/mantidproject/mantid/blob/master/MantidPlot/src/Mantid/MantidUI.cpp#L227). For GSAS specific stuff I'd suggest the main window constructor of the engineering GUI?

Reply:
@samueljackson92 I agree but I've decided to put them in the constructor for the model - I don't see any reason for any aspects of the GUI apart from the GSAS model to care that we're passing GSASII-related objects in signals. Is this OK?

Reply:
Yes that is more sensible